black-jack-redo.rb

```diff
@@ -121,65 +238,50 @@ def view_deck(array)
 	input = gets.chomp
 	input.downcase!      
 
-	if input == 'd' && cards == 52
-		#print_with_stars()
-		puts 'Your were dealt: '
-		user.push(array.pop)
-		user.push(array.pop)
-		puts array.length
-		puts user.length
-		#value = handcount(array)
-		#puts "Your cards add up to #{value} points!"
-		cards = 50
-		#format
-		puts '-------'
-		gameloop(array, user, cards)
-	elsif input == 'd' && cards != 52
-		#print_with_stars()
-		count4 = handcount(array)
-		ha, bool = drawone(array, cards, count4)
-		user.push(ha)
-		#user.push(drawone(array, cards, count4))
-		cards = cards - 1
-		#format
-		puts '-------'
+	if input == 'd' && array.length == 52
+		user_hand, bool = drawone(array, user)
+		user_hand, bool = drawone(array, user)
+		dealer_hand, bool = drawone(array, dealer)
+		dealer_hand, bool = drawone(array, dealer)
+		string = ''
+		string_append, hand_value = viewhand(user, 'true')
+		string = string + string_append + 'The dealer has a ' + get_card(dealer[0])
+		print_with_stars(string)
+		gameloop(array, user, dealer)
+	elsif input == 'd' && array.length != 52
+		drawn_card = array[array.length-1]
+		updated_hand, bool = drawone(array, user)
+		string = 'Your were dealt: ' + get_card(drawn_card)
+		string = string + updated_hand
+
 		if bool == true
-			array = []
+			print_with_stars(string + "Oh no! Your total is now #{get_value_of_hand(user)} which is greater than 21, you lose!")
 			user = []
-			cards = 52
-			array = array_gen(array)
-			gameloop(array, user, cards)
+			dealer = []
+			gameloop(array_gen, user, dealer)
 		else
-			gameloop(array, user, cards)
+			print_with_stars(string)
+			gameloop(array, user, dealer)
 		end
-	elsif input == 'v' && cards != 52
-		string = "You have #{52-cards} cards in your hand! " + "\n" + "Your hand includes the following:" + "\n"
-		count4, string2 = viewhand(array, string)
-		string = string2 + "Your cards add up to #{count4} points!" + "\n"
-		print_with_stars(string)
-		puts '-------'
-		gameloop(array, user, cards)
-	elsif input == 's' && cards != 52
-		string = "You have #{52-cards} cards in your hand! " + "\n" + "Your hand includes the following:" + "\n"
-		count4, string = viewhand(array, string)
-		string += dealer(count4)
+	elsif input == 'v' && array.length != 52
+		string, hand_value = viewhand(user, 'true')
 		print_with_stars(string)
-		array = []
+		gameloop(array, user, dealer)
+	elsif input == 's' && array.length != 52
```

Comment:
Refactor the logic under each if statement to call a method. For example, if `input == 's'` then run a method called `player_stands`.